func.go

```diff
@@ -169,8 +169,16 @@ func RegisterFunc(fptr any, cfn uintptr) {
 					stack++
 				}
 			case reflect.Struct:
-				if runtime.GOOS != "darwin" || (runtime.GOARCH != "amd64" && runtime.GOARCH != "arm64") {
-					panic("purego: struct arguments are only supported on darwin amd64 & arm64")
+				// Struct arguments are supported on:
+				// - darwin amd64 & arm64
+				// - linux amd64 & arm64
```

Comment:
I think this comment is redundant. Create a utility function like `isStructSupportedForRegisterFunc` and use it.